src/test/java/picard/sam/CompareSAMsTest.java

```diff
@@ -120,4 +125,77 @@ public void testComparisons(final String f1, final String f2, final ArrayList<St
+                {"unmapped_first.sam", "unmapped_first.sam", new Object[][] { {"0,0", 1}, {"30,30", 1}}},
+                {"genomic_sorted.sam", "genomic_sorted_sam_v1.6.sam", new Object[][] { {"20,20", 1}, {"30,30", 1}}},
+                {"unsorted.sam", "unsorted.sam", new Object[][] { {"20,20", 1}, {"30,30", 1}}},
+                {"unsorted.sam", "unsorted2.sam", new Object[][] { {"20,20", 1}}},
+                {"duplicate_base.sam", "duplicate_four_mismatch_strict.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base.sam", "duplicate_four_mismatch_lenient_one_align_differ.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base.sam", "duplicate_two_mismatch_lenient.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base.sam", "duplicate_four_mismatch_lenient.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base.sam", "duplicate_four_mismatch_strict.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base_queryname.sam", "duplicate_four_mismatch_strict_queryname.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base_queryname.sam", "duplicate_four_mismatch_lenient_one_align_differ_queryname.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base_queryname.sam", "duplicate_two_mismatch_lenient_queryname.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base_queryname.sam", "duplicate_four_mismatch_lenient_queryname.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"duplicate_base_queryname.sam", "duplicate_four_mismatch_strict_queryname.sam", new Object[][] { {"20,20", 2}, {"30,30", 12}}},
+                {"genomic_sorted.sam", "mq0_2.sam", new Object[][] { {"20,0", 1}, {"30,30", 1}}},
+                {"mq0_1.sam", "mq0_2.sam", new Object[][] { {"0,0", 1}, {"30,30", 1}}}
+        };
+    }
+
+    @Test(dataProvider = "compareSamsMQConcordance")
+    public void testMQConcordance(final String f1, final String f2, final Object[][] expectedMQConcordance) throws IOException {
+        final Path tmpOutput = Files.createTempFile("compareSam", ".tsv");
+        final Path tmpMQConcordanceOutput = Files.createTempFile("compareSam.mqconcordance", ".tsv");
+        final String in1 = new File(TEST_FILES_DIR, f1).getAbsolutePath();
+        final String in2 = new File(TEST_FILES_DIR, f2).getAbsolutePath();
+        final ArrayList<String> commandArgs = new ArrayList<>(
+                Arrays.asList(
+                        in1,
+                        in2,
+                        "O=" + tmpOutput,
+                        "OUTPUT_MQ_CONCORDANCE=" + tmpMQConcordanceOutput
+                )
+        );
+        // If the files cannot be compared (e.g. if their sort order differs) we expect an exception to be thrown.
+        if (expectedMQConcordance == null) {
```

Comment:
the tool shouldn't throw exception in this case, but in future you can replace this pattern with `Assert.assertThrows`